pkg/events/events.go

```diff
@@ -145,7 +145,10 @@ func New(ctx context.Context, name string, identifiers CombinedIdentifiers, data
 		data: data,
 	}
 	if data, ok := data.(interface{ GetCorrelationIDs() []string }); ok {
-		evt.innerEvent.CorrelationIDs = append(evt.innerEvent.CorrelationIDs, data.GetCorrelationIDs()...)
+		if cids := data.GetCorrelationIDs(); len(cids) > 0 {
+			ctx := ContextWithCorrelationID(ctx, cids...)
```

Comment:
Please revert this change, having a dedicated function was good, but it should not be exported.